src/retrieval/inverted_index.h

```diff
@@ -379,22 +407,18 @@ void InvertedIndex<kDescType, kDescDim,
     inverted_file.ComputeIDFWeight(image_ids.size());
   }
 
-  const int max_image_id =
-      *std::max_element(image_ids.begin(), image_ids.end());
-
-  std::vector<double> self_similarities(max_image_id + 1, 0.0);
+  std::unordered_map<int, double> self_similarities;
```

Comment:
Since the number of images is known, ``self_similarities`` could be preallocated.

Reply:
Yeah -- "std::unordered_map<int, double> self_similarities(image_ids.size())" should do the trick?

Reply:
Or .reserve()